### DISCUSSION TITLE
Should Sakura Editor support Traditional Chinese and Simplified Chinese File Encoding or even other Asian Language Encoding?

### DISCUSSION THREAD

**Discussion by asdf913** [sakura-editor/sakura#1916] [General]
Open source application should be world wide and there may be an occasion to handle file encoded in Big-5 (Traditional Chinese) or GB (Simplified Chinese), it would be much better support Chinese encoding and even other Asian Language Encoding (i.e. Korean or Thai)..

![image](https://user-images.githubusercontent.com/2301325/224201063-40a9c2a9-c262-4f58-aea5-9ae7489ea5b7.png)
![image](https://user-images.githubusercontent.com/2301325/224201004-e3c2eaf9-2bb7-4981-8044-659fa9119989.png)

**Comment by kasumikagari** [comment]
These features are nice-to-have things without a doubt. It boils down to the lack of interest from the dev team and our contributors as they're mostly comprised of Japanese living in Japan.

We've occasionally got similar requests regarding East Asian languages from time to time, but I actually haven't heard much about the sakura users outside of Japan. Can you inform us about that? Is there a sizable number of users? How much of interest in additional language support?

**Comment by kasumikagari** [comment]
See also: #1782

**Comment by arigayas** [comment]
See also: #1369

**Comment by beru** [comment]
@asdf913 Sakura editor's built-in character encoding detection feature supports limited sets of character encodings but external libraries such as ICU4C and uchardet can be used to enhance the feature.

ICU4C : https://github.com/sakura-editor/sakura/pull/1104
uchardet : https://github.com/sakura-editor/sakura/pull/1726#pullrequestreview-764835670

If you're still interested in this and not sure how to use the libraries, I would write instructions for you.
